File: KnowledgeSpaceApk/adminapp/src/main/java/com/example/adminapp/GroupFragAdm/GrpQuizesAdm/adapterRecVQuizesMainScAdm.java
package com.example.adminapp.GroupFragAdm.GrpQuizesAdm;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.adminapp.R;

import java.util.ArrayList;

public class adapterRecVQuizesMainScAdm extends RecyclerView.Adapter<adapterRecVQuizesMainScAdm.myviewholder> {
    ArrayList<dataModelRecVQuizesMainScAdm> dataHolder;
    ArrayList<dataModelRecVQuizesMainScAdm> dataHolderBackup;

    public adapterRecVQuizesMainScAdm(ArrayList<dataModelRecVQuizesMainScAdm> dataHolder) {
        this.dataHolder = dataHolder;
        dataHolderBackup = new ArrayList<>(dataHolder);
    }

    @NonNull
    @Override
    public myviewholder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new myviewholder(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.single_row_des_recv_quizes_mainsc_adm,parent,false));
    }

    @Override
    public void onBindViewHolder(@NonNull myviewholder holder, int position) {
        holder.quizeTitleSingleRdsQuizesMainScAdm.setText(dataHolder.get(position).getQuizeTitleSingleRdsQuizesMainScAdm());
        holder.quizeDurationSingleRdsQuizesMainScAdm.setText(new StringBuffer().append("Quiz Duration : ").append(dataHolder.get(position).getQuizeDurationSingleRdsQuizesMainScAdm()));
        holder.totalNoQuestionsSingleRdsQuizesMainScAdm.setText(new StringBuffer().append("Total Questions : ").append(dataHolder.get(position).getTotalNoQuestionsSingleRdsQuizesMainScAdm()));
    }

    @Override
    public int getItemCount() {
        return dataHolder.size();
    }

    public class myviewholder extends RecyclerView.ViewHolder{

        TextView quizeTitleSingleRdsQuizesMainScAdm,quizeDurationSingleRdsQuizesMainScAdm,totalNoQuestionsSingleRdsQuizesMainScAdm;

        public myviewholder(@NonNull View itemView) {
            super(itemView);
            quizeTitleSingleRdsQuizesMainScAdm = itemView.findViewById(R.id.quizeTitleSingleRdsQuizesMainScAdm);
            quizeDurationSingleRdsQuizesMainScAdm = itemView.findViewById(R.id.quizeDurationSingleRdsQuizesMainScAdm);
            totalNoQuestionsSingleRdsQuizesMainScAdm = itemView.findViewById(R.id.totalNoQuestionsSingleRdsQuizesMainScAdm);
        }
    }
}

File: KnowledgeSpaceApk/adminapp/src/main/java/com/example/adminapp/GroupFragAdm/GrpQuizesAdm/adapterRecVQuizesMainScAdm.kt
package com.example.adminapp.GroupFragAdm.GrpQuizesAdm

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.adminapp.R

class adapterRecVQuizesMainScAdm(var dataHolder: ArrayList<dataModelRecVQuizesMainScAdm>?) :
    RecyclerView.Adapter<adapterRecVQuizesMainScAdm.myviewholder>() {
    var dataHolderBackup: ArrayList<dataModelRecVQuizesMainScAdm>

    init {
        dataHolderBackup = ArrayList(dataHolder)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): myviewholder {
        return myviewholder(
            LayoutInflater.from(parent.context)
                .inflate(R.layout.single_row_des_recv_quizes_mainsc_adm, parent, false)
        )
    }

    override fun onBindViewHolder(holder: myviewholder, position: Int) {
        holder.quizeTitleSingleRdsQuizesMainScAdm.text =
            dataHolder!![position]. quizeTitleSingleRdsQuizesMainScAdm
        holder.quizeDurationSingleRdsQuizesMainScAdm.text =
            StringBuffer().append("Quiz Duration : ").append(
                dataHolder!![position].quizeDurationSingleRdsQuizesMainScAdm
            )
        holder.totalNoQuestionsSingleRdsQuizesMainScAdm.text =
            StringBuffer().append("Total Questions : ").append(
                dataHolder!![position].totalNoQuestionsSingleRdsQuizesMainScAdm
            )
    }

    override fun getItemCount(): Int {
        return dataHolder!!.size
    }

    inner class myviewholder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var quizeTitleSingleRdsQuizesMainScAdm: TextView
        var quizeDurationSingleRdsQuizesMainScAdm: TextView
        var totalNoQuestionsSingleRdsQuizesMainScAdm: TextView

        init {
            quizeTitleSingleRdsQuizesMainScAdm =
                itemView.findViewById(R.id.quizeTitleSingleRdsQuizesMainScAdm)
            quizeDurationSingleRdsQuizesMainScAdm =
                itemView.findViewById(R.id.quizeDurationSingleRdsQuizesMainScAdm)
            totalNoQuestionsSingleRdsQuizesMainScAdm =
                itemView.findViewById(R.id.totalNoQuestionsSingleRdsQuizesMainScAdm)
        }
    }
}
